docs/installation/advanced/integrating-from-source-webpack.md

```diff
@@ -359,32 +360,32 @@ If you want to use TypeScript, the `ckeditor` file looks very similar. However,
 ```ts
 // ckeditor.ts
 
-import ClassicEditorBase from '@ckeditor/ckeditor5-editor-classic/src/classiceditor';
-import EssentialsPlugin from '@ckeditor/ckeditor5-essentials/src/essentials';
-import AutoformatPlugin from '@ckeditor/ckeditor5-autoformat/src/autoformat';
-import BoldPlugin from '@ckeditor/ckeditor5-basic-styles/src/bold';
-import ItalicPlugin from '@ckeditor/ckeditor5-basic-styles/src/italic';
-import BlockQuotePlugin from '@ckeditor/ckeditor5-block-quote/src/blockquote';
-import HeadingPlugin from '@ckeditor/ckeditor5-heading/src/heading';
-import LinkPlugin from '@ckeditor/ckeditor5-link/src/link';
-import ListPlugin from '@ckeditor/ckeditor5-list/src/list';
-import ParagraphPlugin from '@ckeditor/ckeditor5-paragraph/src/paragraph';
-
-export default class ClassicEditor extends ClassicEditorBase {}
-
-ClassicEditor.builtinPlugins = [
-    EssentialsPlugin,
-    AutoformatPlugin,
-    BoldPlugin,
-    ItalicPlugin,
-    BlockQuotePlugin,
-    HeadingPlugin,
-    LinkPlugin,
-    ListPlugin,
-    ParagraphPlugin
+import { ClassicEditor } from '@ckeditor/ckeditor5-editor-classic';
+import { Essentials } from '@ckeditor/ckeditor5-essentials';
+import { Autoformat } from '@ckeditor/ckeditor5-autoformat';
+import { Bold } from '@ckeditor/ckeditor5-basic-styles';
+import { Italic } from '@ckeditor/ckeditor5-basic-styles';
+import { BlockQuote } from '@ckeditor/ckeditor5-block-quote';
+import { Heading } from '@ckeditor/ckeditor5-heading';
+import { Link } from '@ckeditor/ckeditor5-link';
+import { List } from '@ckeditor/ckeditor5-list';
+import { Paragraph } from '@ckeditor/ckeditor5-paragraph';
+
+export default class CustomEditor extends ClassicEditor {}
+
+CustomEditor.builtinPlugins = [
```

Comment:
This isn't consistent with how we define builds today. We had to change the syntax a bit.

Reply:
I meant this: [https://github.com/ckeditor/ckeditor5/blob/29857dbdec67595ab35500d4f99c1883535e40dd/packages/ckeditor5-build-inline/src/ckeditor.ts#L29-L30](https://github.com/ckeditor/ckeditor5/blob/29857dbdec67595ab35500d4f99c1883535e40dd/packages/ckeditor5-build-inline/src/ckeditor.ts#L29-L30)

AFAIR (cc @niegowski @filipsobol @arkflpc ) we changed this recently for a particular reason. But I don't remember what and whether it applies here too.

Reply:
This was changed in builds to allow inferring `builtinPlugins` property type.

Without that, the type would be `Array<PluginConstructor>`.

With type inference, it is something like `Array<typeof Essentials | typeof Autoformat | typeof Bold /* etc. */>`.

Thanks to it, in `.d.ts` of this build, all plugin imports are included (because now they are used as part of `ClassicEditor` type). Because of it, all imports are seen by client code (including `./augmentation.ts` files) and the type inference of `EditorConfig` (and commands) works properly now.